src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -659,21 +659,24 @@ public void Run()
                             {
                                 EnsureThreadAllocationStats();
 
-                                using (context.OpenReadTransaction())
                                 using (var scope = new DisposableScope())
                                 using (var merged = new ExtractedItemsEnumerator<TExtracted>(stats))
                                 {
                                     var nextEtag = loadLastProcessedEtag + 1;
 
-                                    Extract(context, merged, nextEtag, EtlItemType.Document, stats, scope);
+                                    List<TTransformed> transformations;
+                                    using (context.OpenReadTransaction())
+                                    {
+                                        Extract(context, merged, nextEtag, EtlItemType.Document, stats, scope);
 
-                                    if (ShouldTrackCounters())
-                                        Extract(context, merged, nextEtag, EtlItemType.CounterGroup, stats, scope);
+                                        if (ShouldTrackCounters())
+                                            Extract(context, merged, nextEtag, EtlItemType.CounterGroup, stats, scope);
                                     
-                                    if (ShouldTrackTimeSeries())
-                                        Extract(context, merged, nextEtag, EtlItemType.TimeSeries, stats, scope);
+                                        if (ShouldTrackTimeSeries())
+                                            Extract(context, merged, nextEtag, EtlItemType.TimeSeries, stats, scope);
 
-                                    var transformations = Transform(merged, context, stats, state);
+                                        transformations = Transform(merged, context, stats, state);
+                                    }
```

Comment:
1. I don't think we can do that. If there is no transformation script we just put a document as is. So we need to have the read transaction open while sending data to a destination:

https://github.com/ravendb/ravendb/blob/18a605ad027854aae791f060265cd21e30d064e7/src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs#L277

2. How this change could cause any regression - https://issues.hibernatingrhinos.com/issue/RavenDB-15713 - since we have it from the beginning?

Reply:
Now I understand how you found https://github.com/ravendb/ravendb/pull/11385. Because tests here are causing this. I believe because you're tying to access data after disposing the read transaction

Reply:
You are correct on both comments :-)
Still investigating..